lib/chargebee_rails/sync_plans.rb

```diff
@@ -2,14 +2,14 @@ module ChargebeeRails
   class SyncPlans
     attr_accessor :messages
 
-  	def self.sync
+  	def self.sync(create:true, update:true, delete:false)
   		syncer = SyncPlans.new
-  		return syncer.do_sync
+  		return syncer.do_sync(create:create, update:update, delete:delete)
```

Comment:
@ConfusedVorlon Is it possible to refactor/cleanup passing create, update and delete attrs ?

Reply:
I'm not sure what you mean. Can you elaborate?

Reply:
We felt that the params being passed around looks less clean. Do you think we can have the create, update and delete options as attr_accessors and then initialize them with syncer and access them where needed as the object's attribute ?

Reply:
you don't have to pass them. They all have defaults.

just call syncer.do_sync() if you want default behaviour

if you want to turn off the update, then call 
syncer.do_sync(update:false)

I think thats cleaner than 

syncer.update = false
syncer.do_sync()

Reply:
I was referring to the SyncPlans class like so,

`
module ChargebeeRails
  class SyncPlans
    attr_accessor :messages, :create, :update, :delete
    
    def self.sync(create:true, update:true, delete:false)
      syncer = SyncPlans.new(create: create, update: update, delete: delete)
      syncer.get_plans
      syncer.sync_plans
      return syncer.messages
    end

    ...

    def sync_plans
      output "Removed #{remove_plans.count} plan(s)" if (self.delete)
      output "Created #{create_new_plans.count} plan(s)" if (self.create)
      output "Updated all #{update_all_plans.count} plan(s)" if (self.update)
    end
  end
end
` 
Let us know what you think.

Reply:
that's fine by me. Can you make the change?

Reply:
@ConfusedVorlon we will make the changes regarding the code refactoring, can you please fix the conflict mentioned here ?